src/app/editor/searchable-dropdown/searchable-dropdown.component.spec.ts

```diff
@@ -0,0 +1,109 @@
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+*/
+
+import { provide } from '@angular/core';
+import { TestComponentBuilder } from '@angular/compiler/testing';
+import {
+  ComponentFixture,
+  inject
+} from '@angular/core/testing';
+
+import { SearchableDropdownComponent } from './searchable-dropdown.component';
+
+import { DROPDOWN_DIRECTIVES } from 'ng2-bootstrap/ng2-bootstrap';
+import { FilterByPrefixPipe } from '../shared/pipes';
+
+// fixed value returned by the pipe.
+const filterResults = ['FilteredByPrefix'];
+
+class MockFilterByPrefixPipe extends FilterByPrefixPipe {
+
+  transform(array: Array<string>, prefix: string): Array<string> {
+    return filterResults;
+  }
+}
+
+describe('SearchableDropdownComponent', () => {
+
+  const providers = [...DROPDOWN_DIRECTIVES,
+    provide(FilterByPrefixPipe, { useClass: MockFilterByPrefixPipe })];
+  let fixture: ComponentFixture<SearchableDropdownComponent>;
+  let component: SearchableDropdownComponent;
+  let nativeEl: HTMLElement;
+  let inputEl: HTMLInputElement;
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+    fixture = tcb
+      .overrideProviders(SearchableDropdownComponent, providers)
+      .createSync(SearchableDropdownComponent);
+    component = fixture.componentInstance;
+    component.value = 'default';
+    component.items = [];
+    fixture.detectChanges();
+
+    nativeEl = fixture.nativeElement;
+    inputEl = nativeEl
+      .querySelector('input, textarea') as HTMLInputElement;
+  }));
+
+  it('should map shortcut to correct value when pressed enter', () => {
+    component.shortcutMap = {
+      s: 'shortcut'
+    };
+
+    // input change, to update component.prefix;
+    inputEl.value = 's';
+    inputEl.dispatchEvent(new Event('input'));
+    fixture.detectChanges();
+
+    // trigger Enter key press
+    let enterPressedEvent = new Event('keypress');
+    enterPressedEvent['key'] = 'Enter';
+    inputEl.dispatchEvent(enterPressedEvent);
+
+    expect(component.value).toEqual('shortcut');
+  });
+
+  it('should show all if prefix is empty', () => {
+    // set enum.items
+    let items = ['First', 'Second'];
+    component.items = items;
+    fixture.detectChanges();
+
+    let dropdownItems = nativeEl.querySelectorAll('.dropdown-item');
+    for (let i = 0; i < dropdownItems.length; i++) {
+      expect(dropdownItems.item(i).textContent).toEqual(items[i]);
+    }
+  });
+
+  it('should show filtered items by pipe if prefix is not empty', () => {
+    inputEl.value = 'anything';
+    inputEl.dispatchEvent(new Event('input'));
+    fixture.detectChanges();
+
+    let items = filterResults;
+    let dropdownItems = nativeEl.querySelectorAll('.dropdown-item');
+    for (let i = 0; i < dropdownItems.length; i++) {
+      expect(dropdownItems.item(i).textContent).toEqual([i]);
```

Comment:
You can have look at dropdownItems defined at `L104`, since `MockFilterByPrefixPipe` is fixed, it always returns the same results. The idea is to test when something put into input if it shows results that returned from the pipe.

We test whether if pipe returns the correct results, in its own test.

However this test has some issues so I will update it soon.